nav2_system_tests/src/costmap_filters/test_keepout_launch.py

```diff
@@ -0,0 +1,133 @@
+
+    # Replace the `use_astar` setting on the params file
+    param_substitutions = {
+        'planner_server.ros__parameters.GridBased.use_astar': os.getenv('ASTAR'),
+        'filter_mask_server.ros__parameters.yaml_filename': filter_mask_file,
+        'yaml_filename': filter_mask_file}
+    configured_params = RewrittenYaml(
+        source_file=params_file,
+        root_key='',
+        param_rewrites=param_substitutions,
+        convert_types=True)
+
+    return LaunchDescription([
+        SetEnvironmentVariable('RCUTILS_CONSOLE_STDOUT_LINE_BUFFERED', '1'),
+
+        # Launch gazebo server for simulation
+        ExecuteProcess(
+            cmd=['gzserver', '-s', 'libgazebo_ros_init.so',
+                 '--minimal_comms', world],
+            output='screen'),
+
+        # TODO(orduno) Launch the robot state publisher instead
+        #              using a local copy of TB3 urdf file
+        Node(
+            package='tf2_ros',
+            executable='static_transform_publisher',
+            output='screen',
+            arguments=['0', '0', '0', '0', '0', '0', 'base_footprint', 'base_link']),
+
+        Node(
+            package='tf2_ros',
+            executable='static_transform_publisher',
+            output='screen',
+            arguments=['0', '0', '0', '0', '0', '0', 'base_link', 'base_scan']),
+
+        Node(
+            package='nav2_lifecycle_manager',
+            executable='lifecycle_manager',
+            name='lifecycle_manager_filters',
+            output='screen',
+            parameters=[{'node_names': ['filter_mask_server', 'costmap_filter_info_server']},
+                        {'autostart': True}]),
+
+        # Nodes required for Costmap Filters configuration
+        Node(
+            package='nav2_map_server',
+            executable='map_server',
+            name='filter_mask_server',
+            output='screen',
+            parameters=[configured_params]),
+
+        Node(
+            package='nav2_map_server',
+            executable='costmap_filter_info_server',
+            name='costmap_filter_info_server',
+            output='screen',
+            parameters=[configured_params]),
+
+        IncludeLaunchDescription(
+            PythonLaunchDescriptionSource(
+                os.path.join(bringup_dir, 'launch', 'bringup_launch.py')),
+            launch_arguments={'namespace': '',
+                              'use_namespace': 'False',
+                              'map': map_yaml_file,
+                              'use_sim_time': 'True',
+                              'params_file': configured_params,
+                              'bt_xml_file': bt_navigator_xml,
+                              'autostart': 'True'}.items()),
+    ])
+
+
+def main(argv=sys.argv[1:]):
+    ld = generate_launch_description()
+
+    test1_action = ExecuteProcess(
+        cmd=[os.path.join(os.getenv('TEST_DIR'), 'tester_node.py'),
+             '-r', '-2.0', '-0.5', '0.0', '2.0'],
```

Comment:
Do these points actually go through/around that keepout zone?

Reply:
Yes, the starting point "A" and finish "B" was remained as in `nav2_system_tests/src/system` tests, and the keepout zone was selected along the path as follows from the picture below:
![plan_description](https://user-images.githubusercontent.com/60094858/95823769-de114180-0d36-11eb-963a-8a0b9d4abeff.png)

Reply:
Maybe, I'd better to attach this picture to testcase as well?

Reply:
Hi, an image would be helpful, yes. I think you should update B to be straight up the aisle way from A. That I'm afraid of is that some small change in initial conditions will make this test not really do its job. Since both paths are close-ish to the same length, its not immediately clear to me that 100% of the time that the "actual path" wouldn't also be the path regardless of the keepout zone. 

If we make the A-B in a straight line up the aisle, then we can add a keepout block in the aisle between 2 of the pillars to force the robot into the left aisle or around the edge. That's a clear case of keepout working because it would never do that with a value straight line path.

tl;dr, I think this would be more robust to changes into the future if we:
- Moved B to straight up from A
- Moved the keepout between 2 pillars in the straightline path

Reply:
Okay, done in https://github.com/ros-planning/navigation2/pull/2030/commits/8d12009621566b955c0bd1ec0be14fe93154c3c8